src/edu/csus/ecs/pc2/core/execute/ExecuteTimer.java

```diff
@@ -116,61 +109,28 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
             }
 
         }
-
         // update the on-screen display time
-        getTimerCountLabel().setText(newTime);
-
-        if (e.getSource() == getbtnTerminate()) {
-            connEtoC1(e);
+        if(usingGUI) {
+            ivjExecuteTimerFrame.setTimerCountLabelText(newTime);
         }
     }
 
     /**
      * This terminates the process that this timer is working
      */
-    public void btnTerminateActionPerformed(java.awt.event.ActionEvent actionEvent) {
-
-        if (doAutoStop) {
-            log.config("ExecuteTimer - User hit terminate while AJ'ing.");
-            setOtherContactStaff(true);
-        }
-
-        stopIOCollectors();
-    }
-
-    /**
-     * connEtoC1: (btnTerminate.action.actionPerformed(java.awt.event.ActionEvent) -->
-     * ExecuteTimer.btnTerminate_ActionPerformed(Ljava.awt.event.ActionEvent;)V)
-     *
-     * @param arg1
-     *            java.awt.event.ActionEvent
-     */
-    private void connEtoC1(java.awt.event.ActionEvent arg1) {
+    public void executeFrameTerminated() {
         try {
-            this.btnTerminateActionPerformed(arg1);
+            if (doAutoStop) {
+                log.config("ExecuteTimer - User hit terminate while AJ'ing.");
```

Comment:
I don't understand why the log message above refers to "...while AJ'ing."  It seems to me that the action of invoking this method applies to both AutoJudging AND Manual Judging.  (The method is indeed called when a user presses the "Stop AutoJudging" button on a Judge which is enabled for AutoJuding, but it is _also_ called if a (human) judge presses the "Terminate" button during _manual_  judging.)  I think the message is misleading because it implies this is something tied specifically to use of an "AutoJudge".

Also, it seems to me that `log.config()` is the wrong logging method -- it implies that this message has something to do with "configuration", when the act of the ExecuteTimerFrame being terminated is a result of either the user hitting the "Terminate" button, the user hitting the `Stop AutoJudging` button,  or the system completing the submission execution (unless I've badly misunderstood what the PR changes are intending to do).  Assuming I'm correct, it seems like the logging call should be (something like) `log.info(...)`.  (Perhaps the use of `config()` was based on the concept that the method is being invoked when a user presses the "Stop AutoJudging" button -- but as noted above, from what I can tell that's not the _only_ thing which invokes this method/causes this log message.)

Reply:
Correct.  I guess I was in an "AJ" state of mind when writing this.  
log.config() changed to log.info().

Fixed in next push.